lib/file_builder.ex

```diff
@@ -5,57 +5,74 @@ defmodule Floorplan.FileBuilder do
   Takes a list of UrlLink structs and writes to file
   """
 
-  alias Floorplan.FileCounter
-  alias Floorplan.FileList
   alias Floorplan.Utilities
 
-  def build(url_links) do
-    file_number = FileCounter.increment
-    url_count   = Dict.size(url_links)
-    filename    = if Application.get_env(:floorplan, :test) do
-      "tmp/test_sitemap#{file_number}.xml"
-    else
-      "tmp/sitemap#{file_number}.xml"
-    end
-    FileList.push({filename, :in_progress, url_count})
-
-    case write_url_links_to_file(filename, url_links) do
-      {:ok, :ok} ->
-        {:ok, compressed_filename} = Utilities.compress(filename)
-
-        Logger.info "✓ #{compressed_filename}  -- #{url_count} urls"
-        FileList.replace(filename, {compressed_filename, :completed, url_count})
-      _ ->
-        Logger.info "✕ #{filename}  -- #{url_count} urls"
-        FileList.push({filename, :failed, url_count})
-    end
-
-    if compressed_filename do
-      {:ok, compressed_filename}
-    else
-      {:error, filename}
-    end
+  defmodule Context do
+    defstruct target_directory: "tmp",
+              urls_per_file: 50_000,
+              base_url: "http://www.example.com",
+              sitemap_files: [], # Set by Generator
+              urls: [
+                %Floorplan.Url{location: "/foo/bar.html"}
```

Comment:
It seems potentially problematic to define the defaults with example data instead of just `""`.  If these attributes are not overwritten then we are generating fake data in non-test usage